parts/plugins/x_builddeb.py

```diff
@@ -30,13 +30,18 @@ def patch_snapcraft():
 class XBuildDeb(snapcraft.BasePlugin):
 
     def build(self):
+        super().build()
         self.run(["sudo", "apt-get", "build-dep", "-y", "./"])
         # XXX: get this from "debian/gbp.conf:postexport"
         self.run(["./get-deps.sh"])
-        # run the real build, -ptrue means run "true" to sign the package.
-        # Newer dpkg-buildpackage has "--no-sign" but not the xenial version
-        self.run(["dpkg-buildpackage", "-ptrue"])
+        env=os.environ.copy()
+        if os.getuid() == 0:
+            # disable running the tests during the build when run as root
+            # because quite a few of them will break
+            env["DEB_BUILD_OPTIONS"] = "nocheck"
+        # run the real build
+        self.run(["dpkg-buildpackage"], env=env)
```

Comment:
so we can sign? no need for -ptrue?

Reply:
The right way is to use --no-sign but that is not available in the old version of dpkg-buildpackage in xenial yet. With -ptrue dpkg-buildpackage still fails (exit 1) because it expects a signed $filename.asc) file generated. Without any parameters it will DTRT in a snapcraft environment. On your regular desktop system when running snapcraft it may prompt you for your gpg key to sign the thing which is I why I originally added --no-sign.

Reply:
Would it make sense then to add `-ptrue` if stdin is not a tty then?